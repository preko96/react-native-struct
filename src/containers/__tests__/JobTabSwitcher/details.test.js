import React from 'react'
import { Details } from '../../JobsTabSwitcher/details'
import { Col } from 'native-base'

const getTextValue = wrapper => {
	const firstCol = wrapper.find(Col).first()
	const textComponent = firstCol.children()
	const value = textComponent.children()
	return value.text()
}

it('renders without crashing', () => {
	expect(shallow(<Details/>)).toMatchSnapshot()
})

it('renders primaryStreet', () => {
	const expectedBuildingNumber = 100
	const expectedPrimaryStreet = 'primaryStreet'
	const wrapper = shallow(<Details primaryStreet={expectedPrimaryStreet} buildingNumber={expectedBuildingNumber}/>)
	expect(getTextValue(wrapper)).toEqual(`${expectedBuildingNumber} ${expectedPrimaryStreet}`)
})

it('renders secondaryStreet', () => {
	const expectedValue = 'secondaryStreet'
	const wrapper = shallow(<Details secondaryStreet={expectedValue}/>)
	expect(getTextValue(wrapper)).toEqual(expectedValue)
})

it('renders town', () => {
	const expectedValue = 'town'
	const wrapper = shallow(<Details town={expectedValue}/>)
	expect(getTextValue(wrapper)).toEqual(expectedValue)
})

it('renders country', () => {
	const expectedValue = 'country'
	const wrapper = shallow(<Details country={expectedValue}/>)
	expect(getTextValue(wrapper)).toEqual(expectedValue)
})

it('renders postcode', () => {
	const expectedValue = 'postcode'
	const wrapper = shallow(<Details postcode={expectedValue}/>)
	expect(getTextValue(wrapper)).toEqual(expectedValue)
})

it('renders company', () => {
	const expectedValue = 'company'
	const wrapper = shallow(<Details company={expectedValue}/>)
	expect(getTextValue(wrapper)).toEqual(expectedValue)
})
