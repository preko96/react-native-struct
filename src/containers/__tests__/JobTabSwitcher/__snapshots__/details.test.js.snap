// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<GridNB>
  <ColumnNB />
  <Component
    style={
      Object {
        "backgroundColor": "#d5d5d5",
        "height": 120,
        "left": "50%",
        "position": "absolute",
        "width": 2,
      }
    }
  />
  <ColumnNB
    style={
      Object {
        "marginLeft": 40,
      }
    }
  >
    <Styled(Text)
      numberOfLines={1}
      style={
        Object {
          "lineHeight": 25,
        }
      }
    >
      Reference:
    </Styled(Text)>
    <Styled(Text)
      numberOfLines={1}
      style={
        Object {
          "lineHeight": 25,
        }
      }
    >
      Mobile:
    </Styled(Text)>
    <Styled(Text)
      numberOfLines={1}
      style={
        Object {
          "lineHeight": 25,
        }
      }
    >
      Landline:
    </Styled(Text)>
    <Styled(Text)
      numberOfLines={1}
      style={
        Object {
          "lineHeight": 25,
        }
      }
    >
      Email:
    </Styled(Text)>
  </ColumnNB>
</GridNB>
`;
